modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/checkpoint/CheckpointProgressImpl.java

```diff
@@ -272,6 +272,15 @@ public void nextSnapshot(boolean nextSnapshot) {
         evictedPagesCntr = new AtomicInteger();
     }
 
+    /** {@inheritDoc} */
+    @Override public void clearCounters() {
+        currCheckpointPagesCnt = 0;
```

Comment:
Perhaps it's better to set `currCheckpointPagesCnt` after all other counters, to keep invariant `total pages` >= `written pages` for example. But it's not an issue and it's up to you.